Logger/FileLogger.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.IO;
+
+namespace Logger;
+
+internal class FileLogger : BaseLogger
+{
+    private string? _Path;
+
+    public FileLogger(string? path)
+    {
+        _Path = path;
+    }
```

Comment:
```suggestion
    private string _Path;

    public FileLogger(string path)
    {
        _Path = path ?? throw new ArgumentNullException(path);
    }
```

I would recommend not making this nullable and doing a null check in your constructor to ensure the user isn't forcing null into your constructor.

Reply:
That is likely a better fail-safe solution. We figured that the intent was for FileLoggers to only come from LogFactory, wherein the path is null-checked.

Reply:
That is likely a better fail-safe solution. We figured that the intent was for FileLoggers to only come from LogFactory, wherein the path is null-checked.